### PR TITLE
SFSDK-7 Adds a delete and codebase endpoints

### PR DESCRIPTION
**Pull request by nikgregory**

### CODE REVIEW

File: lib/sfrest/codebase.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+module SFRest
+  # List codebases on the
```
Comment (bnagykekesi) [inline lib/sfrest/codebase.rb]: incomplete sentence

File: lib/sfrest/codebase.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  class Codebase
+    # @param [SFRest::Connection] conn
+    def initialize(conn)
+      @conn = conn
+    end
+
+    # Lists the codebases
+    # @return [Hash] A hash of codebases configured for the factory
```
Comment (bnagykekesi) [inline lib/sfrest/codebase.rb]: sentence ending period

File: lib/sfrest/site.rb [hunk 3]
```diff
@@ -83,13 +83,28 @@ def site_list
     # @param [String] sitename The name of the site to create
     # @param [Integer] group_id  The Id of the group the site is to be a member of
     # @param [String] install_profile The install profile to use when creating the site
-    def create_site(sitename, group_id, install_profile = nil)
+    def create_site(sitename, group_id, install_profile = nil, codebase = nil)
```
Comment (bnagykekesi) [inline lib/sfrest/site.rb]: missing param section for codebase

File: lib/sfrest/site.rb [hunk 4]
```diff
@@ -83,13 +83,28 @@ def site_list
...
+                  'install_profile' => install_profile, 'codebase' => codebase }.to_json
       @conn.post(current_path, payload)
     end
 
+    # Creates a site.
+    # Alias for create_site
+    # @param [String] sitename The name of the site to create
+    # @param [Integer] group_id  The Id of the group the site is to be a member of
+    # @param [String] install_profile The install profile to use when creating the site
+    alias create create_site
```
Comment (bnagykekesi) [inline lib/sfrest/site.rb]: missing param section for codebase

File: spec/site_spec.rb [hunk 5]
```diff
@@ -101,6 +101,19 @@
     end
   end
 
+  describe '#delete' do
+    site_data = generate_site_delete_data
+    site_id = site_data['site_id']
+    site_name = site_data['site']
+    task_id = site_data['task_id']
+    it 'can create a site' do
```
Comment (bnagykekesi) [inline spec/site_spec.rb]: it 'can delete a site' do

File: lib/sfrest/codebase.rb (hunk 1 above)
Comment (nikgregory) [reply lib/sfrest/codebase.rb]: done

File: lib/sfrest/codebase.rb (hunk 2 above)
Comment (nikgregory) [reply lib/sfrest/codebase.rb]: done

**Pull request merged**